modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java

```diff
@@ -1314,21 +1303,87 @@ public void testBindObjectDefinitions() throws Exception {
 			new String[] {"C_A", "C_AA", "C_AAA", "C_AAB"});
 	}
 
+	@Test
+	public void testDeleteCompanyObjectDefinitions() throws Exception {
+		String originalName = PrincipalThreadLocal.getName();
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		Company company = CompanyTestUtil.addCompany();
+
+		PortalInstances.initCompany(company);
+
+		try (SafeCloseable safeCloseable =
+				CompanyThreadLocal.setWithSafeCloseable(
+					company.getCompanyId())) {
+
+			User user = UserTestUtil.getAdminUser(company.getCompanyId());
+
+			Assert.assertNotNull(user);
+
+			PermissionThreadLocal.setPermissionChecker(
+				PermissionCheckerFactoryUtil.create(user));
+
+			PrincipalThreadLocal.setName(user.getUserId());
+
+			ObjectDefinition customObjectDefinition =
+				_objectDefinitionLocalService.addCustomObjectDefinition(
+					user.getUserId(), 0, false, false, false,
+					LocalizedMapUtil.getLocalizedMap("Able"), "Able", null,
+					null, LocalizedMapUtil.getLocalizedMap("Ables"), true,
+					ObjectDefinitionConstants.SCOPE_COMPANY,
+					ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
+					Arrays.asList(
+						ObjectFieldUtil.createObjectField(
+							ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+							ObjectFieldConstants.DB_TYPE_STRING,
+							RandomTestUtil.randomString(),
+							StringUtil.randomId())));
```

Comment:
I believe that's not a good idea because we need to ensure that the user is the user created in the test and the methods of the `ObjectDefinitionTestUtil` are using the `TestPropsValues.getUserId()`. We cannot use the `TestPropsValues.getUserId()` in this case.

I thought about create a new method, but the method will be very similar to the `_objectDefinitionLocalService.addCustomObjectDefinition` call.

See:

```

_objectDefinitionLocalService.addCustomObjectDefinition(
		user.getUserId(), 0, false, false, false,
		LocalizedMapUtil.getLocalizedMap("Able"), "Able", null,
		null, LocalizedMapUtil.getLocalizedMap("Ables"), true,
		ObjectDefinitionConstants.SCOPE_COMPANY,
		ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
		Arrays.asList(
			ObjectFieldUtil.createObjectField(
				ObjectFieldConstants.BUSINESS_TYPE_TEXT,
					ObjectFieldConstants.DB_TYPE_STRING,
					RandomTestUtil.randomString(),
					StringUtil.randomId())));


ObjectDefinitionTestUtil.addCustomObjectDefinition(
		user.getUserId(), 0, false, "Able",
		ObjectDefinitionConstants.SCOPE_COMPANY,
		ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
		_objectDefinitionLocalService,
		Collections.singletonList(
			ObjectFieldUtil.createObjectField(
					ObjectFieldConstants.BUSINESS_TYPE_TEXT,
					ObjectFieldConstants.DB_TYPE_STRING,
					RandomTestUtil.randomString(),
					StringUtil.randomId())));
```